index.html

```diff
@@ -1,185 +1,221 @@
+                                                replaceWord(word) {
+                                                        console.log(this.richText);
+                                                        // replace word 
+                                                        this.richText = this.richText.replaceAll(`<span class="text-error">${word.text}</span>`, word.correct)
+                                                        // remove word from corrected words
+                                                        this.correctedWords = this.correctedWords.filter(correctedWord => correctedWord.correct !== word.correct)
+                                                  
+                                                        console.log(this.richText);
+                                                },
+                                                onInput(event) {
+                                                        const value = event.target.innerText;
+                                                        this.event = event
+                                                        this.text = value;
+                                                },
+                                        },
+                                        watch: {}  
+                                })
+                                // download function
+                                function downloadText() {
+
+                                        let msg = document.getElementById('mainContentBox'); // Get the data from each element on the form.
+                                        let data = msg.innerText;                                         // This variable stores all the data.
+
+                                        // Convert the text to BLOB.
+                                        const textToBLOB = new Blob([data], { type: 'text/plain' });
+                                        const sFileName = 'Download.txt';	   // The file to save the data.
+
+                                        let newLink = document.createElement("a");
+                                        newLink.download = sFileName;
+
+                                        if (window.webkitURL != null) {
+                                                newLink.href = window.webkitURL.createObjectURL(textToBLOB);
+                                        }
+                                        else {
+                                                newLink.href = window.URL.createObjectURL(textToBLOB);
+                                                newLink.style.display = "none";
+                                                document.body.appendChild(newLink);
+                                        }
+
+                                        newLink.click();
+                                }
+
+                                // printbutton function
+                                function printText()

```

Comment:
Well done Harry! I can see that this code here is to handle print!. This is awesome.